### DISCUSSION TITLE
Simple Classifier not working, help needed

### DISCUSSION THREAD

**Discussion by rajsingh100** [Lightning-AI/pytorch-lightning#9249] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
I am trying to train a pytorch lightning simple example but it's giving this error
[](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pl_examples/basic_examples/simple_image_classifier.py)
 
Error:
LitClassifier, MNISTDataModule, seed_everything_default=1234, save_config_overwrite=True, run=False
TypeError: __init__() got an unexpected keyword argument 'run'

pytorch version 1.8.0
pytorch lightning version 1.4.4
conda 4.10.3
Python 3.6.13
OS - Windows.

import torch
from torch.nn import functional as F

import pytorch_lightning as pl
from pl_examples import cli_lightning_logo
from pl_examples.basic_examples.mnist_datamodule import MNISTDataModule
from pytorch_lightning.utilities.cli import LightningCLI

class LitClassifier(pl.LightningModule):
    """
    >>> LitClassifier()  # doctest: +ELLIPSIS +NORMALIZE_WHITESPACE
    LitClassifier(
      (l1): Linear(...)
      (l2): Linear(...)
    )
    """

    def __init__(self, hidden_dim: int = 128, learning_rate: float = 0.0001):
        super().__init__()
        self.save_hyperparameters()

        self.l1 = torch.nn.Linear(28 * 28, self.hparams.hidden_dim)
        self.l2 = torch.nn.Linear(self.hparams.hidden_dim, 10)

    def forward(self, x):
        x = x.view(x.size(0), -1)
        x = torch.relu(self.l1(x))
        x = torch.relu(self.l2(x))
        return x

    def training_step(self, batch, batch_idx):
        x, y = batch
        y_hat = self(x)
        loss = F.cross_entropy(y_hat, y)
        return loss

    def validation_step(self, batch, batch_idx):
        x, y = batch
        y_hat = self(x)
        loss = F.cross_entropy(y_hat, y)
        self.log("valid_loss", loss)

    def test_step(self, batch, batch_idx):
        x, y = batch
        y_hat = self(x)
        loss = F.cross_entropy(y_hat, y)
        self.log("test_loss", loss)

    def configure_optimizers(self):
        return torch.optim.Adam(self.parameters(), lr=self.hparams.learning_rate)

def cli_main():
    cli = LightningCLI(
        LitClassifier, MNISTDataModule, seed_everything_default=1234, save_config_overwrite=True, run=False
    )
    cli.trainer.fit(cli.model, datamodule=cli.datamodule)
    cli.trainer.test(ckpt_path="best")

if __name__ == "__main__":
    cli_lightning_logo()
    cli_main()`

**Comment by ethanwharris** [comment]
Hi @rajsingh100 thanks for the question! The `run` flag was only added to the Lightning CLI in a recent PR and so isn't in a release yet. You have a couple of options:
- Use the Lightning CLI as documented in our `stable` documentation here: https://pytorch-lightning.readthedocs.io/en/stable/common/lightning_cli.html
- Install PL nightly following the instructions here: https://github.com/PyTorchLightning/pytorch-lightning#install-bleeding-edge---future-15

Hope that helps :smiley:

**Reply by rajsingh100** [reply]
Thanks @ethanwharris for your help.
